src/java.desktop/share/native/common/awt/utility/sizecalc.h

```diff
@@ -92,19 +88,19 @@
  * IS_SAFE_... macros to check if the calculations are safe.
  */
 #define SAFE_SIZE_NEW_ARRAY(type, n) \
-    (IS_SAFE_SIZE_MUL(sizeof(type), (n)) ? (new type[(n)]) : throw std::bad_alloc())
+    (IS_SAFE_SIZE_MUL(sizeof(type), (n)) ? (new type[(size_t)(n)]) : throw std::bad_alloc())
 
 #define SAFE_SIZE_NEW_ARRAY2(type, n, m) \
     (IS_SAFE_SIZE_MUL((m), (n)) && IS_SAFE_SIZE_MUL(sizeof(type), (n) * (m)) ? \
```

Comment:
Why we do not cast it here: (n) * (m)?

Reply:
Filed https://bugs.openjdk.org/browse/JDK-8296957

Reply:
You're right, it must be cast here too, I have missed it.

Would you mind submitting a bug?